pkg/executor/simple.go

```diff
@@ -209,9 +209,6 @@ func (e *SimpleExec) setDefaultRoleNone(s *ast.SetDefaultRoleStmt) error {
 	}
 	sql := new(strings.Builder)
 	for _, u := range s.UserList {
-		if u.Hostname == "" {
```

Comment:
The change is picked from the bug fix https://github.com/pingcap/tidb/pull/60082